src/timer_posix.cpp

```diff
@@ -68,21 +70,21 @@ Timer::Timer(const boost::function<void ()>& callback) :
   
   pthread_mutex_init(&this->mutex, NULL);
   pthread_cond_init(&this->cond, NULL);
-  
-  pthread_create(&this->bgthread, NULL, &bg_main_func, this);
 }
 
 Timer::~Timer() {
 
   // Must stop background thread before cleaning up condition variable and
   // mutex. Calling pthread_cond_destroy on a condvar that's being waited
   // on results in undefined behavior--on Fedora 25+ it hangs.
-  pthread_mutex_lock(&this->mutex);
-  this->stopped = true;
-  pthread_cond_signal(&this->cond);
-  pthread_mutex_unlock(&this->mutex);
-
-  pthread_join(this->bgthread, NULL);
+  if (this->bgthread != boost::none) {
+    pthread_mutex_lock(&this->mutex);
+    this->stopped = true;
+    pthread_cond_signal(&this->cond);
+    pthread_mutex_unlock(&this->mutex);
+  
+    pthread_join(*this->bgthread, NULL);
```

Comment:
Is the `*` supposed to be there?

Reply:
Yeah. `bgthread` is a `boost::optional<pthread_t>`, and `pthread_join` wants a `pthread_t`.

Reply:
Ah OK, I see it now.